source/FreeRTOS_IP_Utils.c

```diff
@@ -417,14 +417,25 @@ static BaseType_t prvChecksumProtocolChecks( size_t uxBufferLength,
 
         if( xReturn == 0 )
         {
-            uint8_t ucLength = ( ( ( pxSet->pxProtocolHeaders->xTCPHeader.ucTCPOffset >> 4U ) - 5U ) << 2U );
-            size_t uxOptionsLength = ( size_t ) ucLength;
-            pxSet->uxProtocolHeaderLength = ipSIZE_OF_TCP_HEADER + uxOptionsLength;
-            #if ( ipconfigHAS_DEBUG_PRINTF != 0 )
-                {
-                    pxSet->pcType = "TCP";
-                }
-            #endif /* ipconfigHAS_DEBUG_PRINTF != 0 */
+            uint8_t ucLength = pxSet->pxProtocolHeaders->xTCPHeader.ucTCPOffset >> 4U;
+            size_t uxOptionsLength;
+
+            if( ucLength < 5U )
```

Comment:
ucLength might be less than 0x50, so the value goes wild after calculation.
The first 4 bits of ucTCPOffset must never less than 5 (minimum TCP header size is 20 bytes, which is 5*4).
Add a check here.

Reply:
Can we define a flag for this? somthing like MINIMUM_TCP_SIZE

Reply:
Updated.